geonode/qgis_server/views.py

```diff
@@ -740,3 +742,43 @@ def set_thumbnail(request, layername):
     }
     return HttpResponse(
         json.dumps(retval), content_type="application/json")
+
+@on_ogc_backend(qgis_server.BACKEND_PACKAGE)
+def download_qlr(request, layername):
+    """Download QLR file for a layer.
+
+    :param layername: The layer name in Geonode.
+    :type layername: basestring
+
+    :return: QLR file.
+    """
+    # execute only in qgis_server
+    if 'geonode.qgis_server' in settings.INSTALLED_APPS:
+        layer = get_object_or_404(Layer, name=layername)
+        ogc_url = reverse('qgis_server:layer-request',
+                          kwargs={'layername': layername})
+        url = settings.SITEURL + ogc_url.replace("/", "", 1)
+
+        layers = [{
+            'type': 'raster',
+            'display': layername,
+            'driver': 'wms',
+            'crs': 'EPSG:4326',
+            'format': 'image/png',
+            'styles': '',
+            'layers': layer.title,
+            'url': url
+            }]
+        json_layers = json.dumps(layers)
+
+        url_server = settings.QGIS_SERVER_URL + \
+            '?SERVICE=LAYERDEFINITIONS&LAYERS=' + json_layers
```

Comment:
Can't we use the `requests` module to generate the querystring? Iike in other views in this file